src/App.tsx

```diff
@@ -1,26 +1,48 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from './app/hooks';
+import { actions as todosActions } from './features/todos';
+import { getTodos } from './api';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const dispatch = useDispatch();
+  const { isLoading } = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    dispatch(todosActions.setLoading(true));
+    getTodos()
+      .then(todosFromServer => {
+        dispatch(todosActions.setTodos(todosFromServer));
+      })
+      .catch(() => {
+        dispatch(todosActions.setError('Error'));
```

Comment:
The error handling in the `catch` block uses a hardcoded error message ('Error'). It might be beneficial to include more details about the error or log the error object to provide better context for debugging.